torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -32,18 +32,38 @@ class GridReductionBuffers : OptOutDispatch {
 
   static std::vector<kir::Allocate*> getSyncAllocs(
       const std::vector<Expr*>& exprs) {
-    GridReductionBuffers fgr;
+    Buffers fgr;
     for (auto expr : exprs) {
       fgr.handle(expr);
     }
     return fgr.sync_allocations_;
   }
 
+  static std::vector<kir::Allocate*> getDynamicAllocs(
```

Comment:
We can remove the static on these interfaces as we may want to infer once and grab both, that way we don't have to re-run the pass.

Something like doing the inference on construction of Buffers (maybe rename to BuffersExtractor or something more descriptive), then have get...Allocs functions just return what we found during that pass.